src/freenet/clients/http/SecurityLevelsToadlet.java

```diff
@@ -698,19 +725,42 @@ public static void generatePasswordFormPage(boolean wasWrong, HTMLNode formNode,
 			content.addChild("#", l10nSec("enterPassword"));
 		}
 
-		formNode.addChild("input",
-		        new String[] { "type", "name", "size" },
-		        new String[] { "password", "masterPassword", "100" });
+		//Creates a table for password prompt and the confirmation box.
+		if (forUpgrade){
+		    HTMLNode table = formNode.addChild("table", "border", "0");
+		    HTMLNode row = table.addChild("tr");
+		    HTMLNode cell = row.addChild("td");
+		    cell.addChild("label", "for", "passwordBox", l10nSec("passwordLabel"));
+		    cell = row.addChild("td");
+		    cell.addChild("input",
+		            new String[] { "id", "type", "name", "size" },
+		            new String[] { "passwordBox", "password", "masterPassword", "100" });
+		    row = table.addChild("tr");
+
+		    row = table.addChild("tr");
+		    cell = row.addChild("td");
+		    cell.addChild("label", "for", "confirmPasswordBox", l10nSec("confirmPasswordLabel"));
+		    cell = row.addChild("td");
+		    cell.addChild("input",
+		            new String[] { "id", "type", "name", "size" },
+		            new String[] { "confirmPasswordBox", "password", "confirmMasterPassword", "100"});
+		}
+		else {
+		    formNode.addChild("label", "for", "passwordBox", l10nSec("passwordLabel"));
+		    formNode.addChild("input",
+		            new String[] { "id", "type", "name", "size" },
+		            new String[] { "passwordBox", "password", "masterPassword", "100" });
```

Comment:
When it was a form with a single field it was less of a deal to duplicate it, but now that it's a table with multiple we should avoid duplicating it. Can you please move generating this to a function - probably one that takes `formNode`?